src/rhsmlib/services/config.py

```diff
@@ -63,79 +68,96 @@ def __setitem__(self, key, value):
+    def __delitem__(self, key: str) -> None:
         self._parser.remove_section(key)
         del self._sections[key]
         if self.auto_persist:
             self.persist()
 
-    def __contains__(self, key):
+    def __contains__(self, key: str) -> bool:
         return key in self._sections
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator:
         return iter(self._parser.sections())
 
-    def iter_sections(self):
+    def iter_sections(self) -> Iterator:
         """An iterator that yields the actual ConfigSection objects instead of just
         the names of the sections."""
         for s in self._parser.sections():
             yield self[s]
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._parser.sections())
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         result = {}
         for name, s in list(self._sections.items()):
             result[name] = repr(s)
         return "%s" % result
 
 
 class ConfigSection(collections.abc.MutableMapping):
-    def __init__(self, wrapper, parser, section, auto_persist=False):
+    """
+    Class representing section in rhsm.conf configuration file
+    """
+
+    def __init__(
+        self, wrapper: Config, parser: rhsm.config.RhsmConfigParser, section: str, auto_persist: bool = False
+    ):
         self._wrapper = wrapper
         self._parser = parser
         self._section = section
         self.auto_persist = auto_persist
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator:
         return iter(self._parser.options(self._section))
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: str) -> str:
         if key in self:
             return self._parser.get(self._section, key)
         raise KeyError("Property '%s' does not exist in section '%s'" % (key, self._section))
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key: str, value: Any) -> None:
         self._parser.set(self._section, key, value)
         if self.auto_persist:
             self._wrapper.persist()
 
-    def __delitem__(self, key):
+    def __delitem__(self, key: str) -> None:
         if key in self:
             self._parser.remove_option(self._section, key)
             if self.auto_persist:
                 self._persist()
         else:
             raise KeyError("Property '%s' does not exist in section '%s'" % (key, self._section))
 
-    def __contains__(self, key):
+    def __contains__(self, key: str) -> bool:
         return self._parser.has_option(self._section, key)
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._parser.options(self._section))
 
-    def _persist(self):
+    def _persist(self) -> None:
         self._wrapper.persist()
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "%s" % self._parser.items(self._section)
 
-    def get_int(self, key):
+    def get_int(self, key: str) -> Union[int, None]:
```

Comment:
I personally prefer `Union[int, None]`, because it explicitly mentions returned types. I think that we can use both variants, because it gives the same hint to IDE.